### PR TITLE
PE-34: Create Radio Button Fieldset.

### PR DESCRIPTION
**Pull request by tbansal-opennms**
### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-34

### CODE REVIEW

File: ui/src/components/Common/RadioButtonFieldSet.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<template>
+<div v-for="category of radioValues" :key="category.key" class="p-field-radiobutton">
+    <RadioButton :id="category.key" :name="name" :value="category.name" v-model="selectedVal" @change="onSelect"/>
+    <label :for="category.key">{{category.name}}</label>
+</div>
+</template> 
+
+<script setup lang="ts">
+
+import {defineEmits, ref, defineProps } from "vue";
```
Comment (mikewrosey) [inline ui/src/components/Common/RadioButtonFieldSet.vue:10]: `defineProps` and `defineEmits` do not need to be imported and are compiled away when <script setup> is processed.
Looks like these imports can also be removed from other components.

File: ui/src/components/Common/Demo/Demo.vue [hunk 2]
```diff
@@ -89,4 +96,16 @@ const cities = ref( [
...
 ]);
+
+const categories = ref([
+            {name: 'Accounting', key: 'A'}, 
+            {name: 'Marketing', key: 'M'}, 
+            {name: 'Production', key: 'P'}, 
+            {name: 'Research', key: 'R'}
+        ]);
+const selectedValue = (val : string) => {
+  console.log('selected value :',val);
```
Comment (mikewrosey) [inline ui/src/components/Common/Demo/Demo.vue:107]: Needs formatting according to the .prettierrc file

File: ui/src/components/Common/RadioButtonFieldSet.vue (hunk 1 above)
Comment (tbansal-opennms) [reply ui/src/components/Common/RadioButtonFieldSet.vue]: Changes are done

File: ui/src/components/Common/Demo/Demo.vue (hunk 2 above)
Comment (tbansal-opennms) [reply ui/src/components/Common/Demo/Demo.vue:107]: Formatting is changed.

**Comment by mikewrosey** [comment]
We'll still need to go through and run the prettier formatter on all these files, but otherwise looks good.

**Review by mikewrosey — APPROVED** [review]

**Pull request merged**